ocaml/xenopsd/lib/topology.ml

```diff
@@ -159,9 +179,23 @@ module NUMA = struct
        [n*multiply ... n*multiply + multiply-1], except we always the add the
        single NUMA node combinations. *)
     (* make sure that single NUMA nodes are always present in the combinations *)
-    let single_nodes =
+    let distance_to_candidate d = (d, float d) in
+    let valid_nodes =
       seq_range 0 (Array.length d)
-      |> Seq.map (fun i -> ((10, 10.0), Seq.return i))
+      |> Seq.filter_map (fun i ->
+             let self_distance = d.(i).(i) in
```

Comment:
There is also a global `self_distance`, perhaps this should use a different name to avoid the confusion.

Reply:
Done